torch/testing/_internal/common_optimizers.py

```diff
@@ -1739,6 +1753,16 @@ def _get_optim_inputs_including_global_cliquey_kwargs(
                 "test_mixed_device_dtype",
                 active_if=TEST_WITH_TORCHDYNAMO,
             ),
+            DecorateInfo(
+                toleranceOverride(
+                    {  # This override is needed because we split addcdiv if we are running in capturable mode
+                        # this results in more casts in fp16 => more error over 7 iterations.
+                        torch.float32: tol(atol=4e-04, rtol=0.01),
```

Comment:
this is pretty large...

Reply:
yeah, not sure if we have other options here, this was in eager and I didn't see any issue with the way I decomposed the addcdiv. This can happen with fp16, as there are fewer casts with addcdiv vs the decomposition I did above.

Reply:
ah did u confirm that if you remove the fp16 Tensors in the params list this wouldn't be necessary?

Reply:
Actually I was able to remove this after all, I think I only needed it in the earlier versions for some reason.